kymatio/scattering2d/backend/numpy_backend.py

```diff
@@ -0,0 +1,178 @@
+# Authors: Edouard Oyallon, Sergey Zagoruyko
+
+import numpy as np
+from collections import namedtuple
+
+BACKEND_NAME = 'numpy'
+
+class Pad(object):
+    def __init__(self, pad_size, input_size, pre_pad=False):
+        """
+            Padding which allows to simultaneously pad in a reflection fashion
+            and map to complex.
+            Parameters
+            ----------
+            pad_size : list of 4 integers
+                size of padding to apply.
+            input_size : list of 2 integers
+                size of the original signal
+            pre_pad : boolean
+                if set to true, then there is no padding, one simply adds the imaginarty part.
+        """
+        self.pre_pad = pre_pad
+        self.pad_size = pad_size
+
+    def __call__(self, x):
+        if self.pre_pad:
+            return x
+        else:
+            np_pad = ((self.pad_size[0], self.pad_size[1]), (self.pad_size[2], self.pad_size[3]))
+            batch_shape = x.shape[:-2]
+            signal_shape = x.shape[-2:]
+            x = x.reshape((-1, 1) + signal_shape)
```

Comment:
Maybe it might be nice to eventually refactor out to core/scattering2d?

Reply:
Not sure that `reshape` is independent of the frontend. If it is, we can move it.

Reply:
I had to put the reshape's here so that the tests would run. Even after adding the reshapes on the numpy frontend, these reshapes here are needed. 

Additionally, the torch frontend has the same reshape operations in the pad call function.

Reply:
Which tests would fail? Which function calls, specifically. I thought this was only called by core `scattering2d()`.

Reply:
This has now been removed (all batch reshaping is done in frontend).